cl/lib/elasticsearch_utils.py

```diff
@@ -584,13 +584,26 @@ def build_has_child_query(
         highlighting_fields = {}
     highlight_options: dict[str, dict[str, Any]] = {"fields": {}}
 
+    fields_to_exclude = []
     for field, fragment_size in highlighting_fields.items():
         number_of_fragments = 0
+        no_match_size = 0
         if fragment_size:
             number_of_fragments = 1
+            # In fields that have a defined fragment size in their HL mapping
+            # e.g., SEARCH_RECAP_CHILD_HL_FIELDS, a 'no_match_size' parameter
+            # is also set. If there are no matching fragments to highlight,
+            # this setting will return a specified amount of text from the
+            # beginning of the field.
+            no_match_size = settings.NO_MATCH_HL_SIZE
+            if not field.endswith("exact"):
+                # The original field is excluded from the response to avoid
+                # returning the entire field from the index.
+                fields_to_exclude.append(field)
```

Comment:
We should refactor this code using conditional expressions to avoid the nested if-statement. For example:

```python
number_of_fragments = 1 if fragment_size else 0
no_match_size = settings.NO_MATCH_HL_SIZE if fragment_size else 0

if fragment_size and not field.endswith("exact"):
    fields_to_exclude.append(field)
```

Reply:
Great,  yeah this makes the code simpler, I've applied the suggestion!